proctor-common/src/main/java/com/indeed/proctor/common/ProctorUtils.java

```diff
@@ -1417,4 +1417,19 @@ private static boolean evaluatePayloadValidator(
 
         return false;
     }
+
+    public static boolean containsUnitlessAllocation(
+            final ConsumableTestDefinition testDefinition,
+            final Map<String, ValueExpression> localContext) {
+        return testDefinition != null
+                && testDefinition.getEnableUnitlessAllocations()
+                && localContext.containsKey("missingExperimentalUnit")
+                && localContext.get("missingExperimentalUnit").getValue(null).equals("true")
```

Comment:
Should we check NPE for "localContext.get("missingExperimentalUnit").getValue(null)" before comparing to "true"

Reply:
Changed it to check localContext.get("missingExperimentalUnit")

and switched to Objects.equals for safer handling of null